tests/core/object/test_object.h

```diff
@@ -36,6 +36,8 @@
 #include "core/object/object.h"
 #include "core/object/script_language.h"
 
+#include "modules/gdscript/gdscript.h"
```

Comment:
This breaks encapsulation. Code outside `modules/gdscript/` shouldn't depend on that module.

So the relevant GDScript test should go in `modules/gdscript/tests/`.

Reply:
Moving the tests to `modules/gdscript/tests` is more tricky than initially expected.
For now I have removed the tests, that verify the correct notification order between native classes and GDScript, so that encapsulation is not broken.

Reply:
Alternatively, since this is test code we don't have to be too strict about itzl, so you could make the whole test and the include depend on `MODULE_GDSCRIPT_ENABLED`.

Reply:
Thanks, I will keep that in mind, if I don't manage to place the tests in `modules/gdscript`.